src/AfricasTalking.php

```diff
@@ -89,6 +82,20 @@ public function __construct($username, $apiKey)
 		]);
 	}
 
+
+	public function base_domain(){
+		define(BASE_DOMAIN, "africastalking.com");
+		define(BASE_SANDBOX_DOMAIN, "sandbox." . self::BASE_DOMAIN);
+
+		if($username === 'sandbox') {
```

Comment:
Hi. I have not tested this so I could entirely be wrong but it seems to me as if `$username` undefined in the function scope. 

I would suggest moving the following lines to above where you're calling the function in `__construct()`

```
$this->username = $username;
```
then change `$username` to `$this->username`